force-app/main/default/classes/DailyIterator.cls

```diff
@@ -1,13 +1,41 @@
 public with sharing class DailyIterator implements RecurrenceIteratorInterface {
+    private List<Datetime> result;
+    private Datetime runningDatetime;
     @TestVisible
-    RecurrenceRule recurrenceRule;
+    private RecurrenceRule rRule;
 
-    public DailyIterator(RecurrenceRule recurrenceRule) {
-        this.recurrenceRule = recurrenceRule;
+    public DailyIterator(RecurrenceRule rRule) {
+        this.rRule = rRule;
     }
 
-    public List<DateTime> iterate(DateTime dateTimeStart) {
-        // TODO: add actual iteration
-        return new List<DateTime>{ dateTimeStart };
+    public List<Datetime> iterate(Datetime datetimeStart) {
+        this.result = new List<Datetime>();
+        this.runningDatetime = datetimeStart;
+        this.rRule = rRule;
+
+        this.result.add(this.runningDatetime);
```

Comment:
is this an opportunity to use the infamous do while loop we talk about in RAD but I've never personally had a cause to use :)

Reply:
I know we use this in javascript a bunch, but I normally only see it in apex if we are shadowing variable names. Since it is technically a preference thing, maybe we can see what others think as well and come to a consensus?

Reply:
Possibly! Happy to think about it further together.

Reply:
I agree with you @jjbennett  on this one. I think just to be consistent with all the other code we have maybe it is better to remove this.